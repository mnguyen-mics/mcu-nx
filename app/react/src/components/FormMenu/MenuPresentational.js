
import React from 'react';
import PropTypes from 'prop-types';
import McsIcons from '../McsIcons';

function MenuPresentational(props) {

  const {
      title,
      subtitles,
      select,
      type,
    } = props;

  return (
    <button className="presentation-item" onClick={select}>
      <div className="content">
        <McsIcons type={type} />
        <div className="title">
          {title}
        </div>
        <div className="subtitle">
          { subtitles && subtitles.map(subtitle => {
            return `${subtitle}, `;
          }) }
        </div>
      </div>
    </button>
  );
}

MenuPresentational.defaultProps = {
  subtitles: null,
  type: 'image',
};

MenuPresentational.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string,
  subtitles: PropTypes.arrayOf(PropTypes.string.isRequired),
  select: PropTypes.func.isRequired,
};

export default MenuPresentational;
