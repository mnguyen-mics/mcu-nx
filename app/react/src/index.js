import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
