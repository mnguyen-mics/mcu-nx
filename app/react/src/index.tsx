/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
