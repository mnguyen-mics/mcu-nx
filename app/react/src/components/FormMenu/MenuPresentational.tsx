
import * as React from 'react';
import McsIcon, { McsIconType } from '../McsIcon';

export interface MenuPresentationalProps {
  title: string;
  type: McsIconType;
  subtitles?: string[];
  select: React.FormEventHandler<any>;
}

const MenuPresentational: React.SFC<MenuPresentationalProps> = ({
  title,
  subtitles,
  select,
  type,
}) => {

  return (
    <button className="presentation-item" onClick={select}>
      <div className="content">
        <div>
          <McsIcon type={type} />
          <div className="title">
            {title}
          </div>
          <div className="subtitle">
            {subtitles && subtitles.map((subtitle, index, arr) => {
              // If it's the last subtitle, the comma is not needed
              if(index === arr.length - 1) {
                return `${subtitle}`;
              } else {
                return `${subtitle}, `;
              }
            })}
          </div>
        </div>
      </div>
    </button>
  );
};

export default MenuPresentational;
