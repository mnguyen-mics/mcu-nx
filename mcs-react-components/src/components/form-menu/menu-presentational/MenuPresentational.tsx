import * as React from 'react';
import McsIcon, { McsIconType } from '../../icon/McsIcon';

export interface MenuPresentationalProps {
  title: string;
  type: McsIconType;
  subtitles?: string[];
  select: React.FormEventHandler<any>;
  disabled?: boolean;
}

const MenuPresentational: React.SFC<MenuPresentationalProps> = ({
  title,
  subtitles,
  select,
  type,
  disabled
}) => {
  return (
    <button className="mcs-menu-presentational" onClick={select} disabled={disabled}>
      <div className="content">
        <McsIcon type={type} />
        <div className="title">{title}</div>
        <div className="subtitle">
          {subtitles &&
            subtitles.map((subtitle, index, arr) => {
              // If it's the last subtitle, the comma is not needed
              if (index === arr.length - 1) {
                return `${subtitle}`;
              } else {
                return `${subtitle}, `;
              }
            })}
        </div>
      </div>
    </button>
  );
};

export default MenuPresentational;
