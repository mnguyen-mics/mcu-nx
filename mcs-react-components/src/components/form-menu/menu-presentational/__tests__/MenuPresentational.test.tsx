import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import MenuPresentational, { MenuPresentationalProps } from '../MenuPresentational';

it('renders the MenuPresentational', () => {
  const props: MenuPresentationalProps = {
    title: 'Title',
    type: 'automation',
    subtitles: ['subtitle_1', 'subtitle_2'],
    select: () => { /* tslint:disable */ console.log('click') /* tslint:enable */ },
  };
  const component = TestRenderer.create(
    <MenuPresentational {...props}/>
  );
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
