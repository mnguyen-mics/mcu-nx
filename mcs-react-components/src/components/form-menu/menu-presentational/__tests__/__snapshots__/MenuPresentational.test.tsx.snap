// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MenuPresentational 1`] = `
<button
  className="mcs-menu-presentational"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <span
      className="mcs-icon "
    >
      <i
        className="mcs-automation"
      />
    </span>
    <div
      className="title"
    >
      Title
    </div>
    <div
      className="subtitle"
    >
      subtitle_1, 
      subtitle_2
    </div>
  </div>
</button>
`;
